var/spack/repos/builtin/packages/cloverleaf/package.py

```diff
@@ -12,21 +12,31 @@ class Cloverleaf(MakefilePackage):
 
     homepage = "http://uk-mac.github.io/CloverLeaf"
     url      = "http://downloads.mantevo.org/releaseTarballs/miniapps/CloverLeaf/CloverLeaf-1.1.tar.gz"
+    git      = "https://github.com/UK-MAC/CloverLeaf.git"
 
     tags = ['proxy-app']
 
+    version('master', tag='master', submodules=True)
     version('1.1', sha256='de87f7ee6b917e6b3d243ccbbe620370c62df890e3ef7bdbab46569b57be132f')
 
     variant('build', default='ref', description='Type of Parallelism Build',
             values=('cuda', 'mpi_only', 'openacc_cray',
                     'openmp_only', 'ref', 'serial'))
+    variant('ieee', default=False, description='Build with IEEE standards')
+    variant('debug', default=False, description='Build with DEBUG flags')
+
+    patch('cloverleaf_aocc.patch', when="@master %aocc", level=0)
```

Comment:
Instead of patching master, why not contribute the patch upstream so that everyone can benefit from it?

Reply:
Plus patches on the main develop branch are doomed to not apply in the future.

Reply:
@adamjstewart , I agree with both of you. 

Currently AOCC supports only one type of build i.e., Reference build.
We are not sure how to apply patch only for that folder inside source that is why we have applied it on master branch.

Any inputs to implement the same will be helpful. 

Thanks.

Reply:
we have raised pull request with CloverLeaf_ref  [158e23d](https://github.com/UK-MAC/CloverLeaf_ref/pull/7) 

@junghans can you please review CloverLeaf pull request !

Reply:
Patch is up streamed to Cloverleaf source, Removed the patch file from package.